src/DDTrace/Integrations/Yii/V2/YiiIntegrationLoader.php

```diff
@@ -0,0 +1,114 @@
+                $span->service = $service;
+
+                if (isset($this->controller->action->actionMethod)) {
+                    $controller = \get_class($this->controller);
+                    $endpoint = "{$controller}::{$this->controller->action->actionMethod}";
+                    $root->setTag("app.endpoint", $endpoint);
+                    $root->setTag(Tag::HTTP_URL, Url::base(true) . Url::current());
+                }
+            }
+        );
+
+        /* Note that Applications are Modules, so this will also trace Application::runAction, but
+         * modules are worth tracing independently, as multiple modules can trigger per
+         * application, such as in the event of an unhandled exception.
+         */
+        \dd_trace_method(
+            'yii\base\Module',
+            'runAction',
+            function (SpanData $span, $args) use ($service) {
+                $span->name = \get_class($this) . '.runAction';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+                $span->resource = isset($args[0]) && \is_string($args[0]) ? $args[0] : $span->name;
+            }
+        );
+
+        \dd_trace_method(
+            'yii\base\Controller',
+            'runAction',
+            function (SpanData $span, $args) use ($service) {
+                $span->name = \get_class($this) . '.runAction';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+                $span->resource = isset($args[0]) && \is_string($args[0]) ? $args[0] : $span->name;
+            }
+        );
+
+        \dd_trace_method(
+            'yii\base\View',
+            'renderFile',
+            function (SpanData $span, $args) use ($service) {
+                $span->name = \get_class($this) . '.renderFile';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+                $span->resource = isset($args[0]) && \is_string($args[0]) ? $args[0] : $span->name;
+            }
+        );
+
+        /* I'm hoping to be able to get app.route.path and proper roote resource name out of this somehow:
+        \dd_trace_method('yii\web\Request', 'resolve', function (SpanData $span, $args, $retval, $ex) use ($service) {
+            $span->name = $span->resource = \get_class($this) . '.resolve';
+            $span->type = Type::WEB_SERVLET;
+            $span->service = $service;
+
+            $pathInfo = $this->getPathInfo();
+
+            if (!$ex) {
+                list($route, $params) = $retval;
+            }
+            return false;
+        });
+        */
+
+        $root->setTag(Tag::SERVICE_NAME, $service);
+        if ('cli' !== PHP_SAPI) {
+            $normalizer = new Urls(explode(',', getenv('DD_TRACE_RESOURCE_URI_MAPPING')));
+            $root->setTag(
+                Tag::RESOURCE_NAME,
+                $_SERVER['REQUEST_METHOD'] . ' ' . $normalizer->normalize($_SERVER['REQUEST_URI']),
```

Comment:
I noticed your commented code above about extracting path info. Do you think it is worth to do it? If not that what you are doing here is done automatically for you in `\DDTrace\Tracer::addUrlAsResourceNameToRootSpan()` at flush time. In general we should do it here in the integration ONLY if we can add more precise info based on the `Route` object. Typically this should be feasible.

Reply:
So far I can extract what the variables should be, so in the case of `/homes/new-york/new-york/manhattan` I can extract `['state' => 'new-york', 'city' => 'new-york', 'neighborhood' => 'manhattan']`, but not where in the URL those values should get plugged. This is why I dropped it, but I thought the commented code is worth keeping around for when we try for V1.

Reply:
I've added support for parameterized routes, both `:` and `?`, in the latest commit. Ready for re-review.